main.cpp

```diff
@@ -77,6 +77,82 @@ returning lhs
 good to go!
 */
 
+#include <iostream>
+struct FloatType 
+{
+    float add( float lhs, float rhs )
+    {
+        return lhs + rhs;
+    }
+    float subtract( float lhs, float rhs )
+    {
+        return lhs - rhs;
+    }
+    float multiply( float lhs, float rhs )
+    {
+        return lhs * rhs;
+    }
+    float divide( float lhs, float rhs )
+    {
+        if(rhs != 0.f) 
+        {
+            return lhs / rhs;
+        }
+        std::cout << std::endl << "warning, floating point division by zero returns 'inf' !" << std::endl;
+        return lhs / rhs;
+    }
+};
+
+struct DoubleType
+{
+    double add( double lhs, double rhs )
+    {
+        return lhs + rhs;
+    }
+    double subtract( double lhs, double rhs )
+    {
+        return lhs - rhs;
+    }
+    double multiply( double lhs, double rhs )
+    {
+        return lhs * rhs;    
+    }
+    double divide( double lhs, double rhs )
+    {
+        if(rhs != 0.0) 
+        {
+            return lhs / rhs;
+        }
+        std::cout << std::endl << "warning, floating point division by zero returns 'inf' !" << std::endl;
+        return lhs / rhs;
+    }
```

Comment:
Don't Repeat Yourself.
rewrite the logic so you only write the `return` statement once.
- [ ] revise